examples/C++/DDS/HelloWorldExampleDiscoveryServer/HelloWorld_main.cpp

```diff
@@ -0,0 +1,244 @@
+                    }
+                    else
+                    {
+                        topic_name = std::string(opt.arg);
+                    }
+                    break;
+
+                case optionIndex::SAMPLES:
+                    if (type == SERVER)
+                    {
+                        print_warning("publisher|subscriber", opt.name);
+                    }
+                    else
+                    {
+                        count = strtol(opt.arg, nullptr, 10);
+                    }
+                    break;
+
+                case optionIndex::INTERVAL:
+                    if (type == PUBLISHER)
+                    {
+                        sleep = strtol(opt.arg, nullptr, 10);
+                    }
+                    else
+                    {
+                        print_warning("publisher", opt.name);
+                    }
+                    break;
+
+                case optionIndex::WAIT:
+                    if (type == PUBLISHER)
+                    {
+                        num_wait_matched = strtol(opt.arg, nullptr, 10);
+                    }
+                    else
+                    {
+                        print_warning("publisher", opt.name);
+                    }
+                    break;
+
+                case optionIndex::LOCATOR:
+                    port = static_cast<uint16_t>(server_address.port);
+
+                    if (regex_match(opt.arg, mr, Arg::ipv4))
+                    {
+                        std::cmatch::iterator it = mr.cbegin();
+                        ip_address = (++it)->str();
+
+                        if ((++it)->matched)
+                        {
+                            int port_int = std::stoi(it->str());
+                            if (port_int <= 65535)
+                            {
+                                port = static_cast<uint16_t>(port_int);
+                            }
+                        }
+                    }
+
+                    if (!ip_address.empty() && port > 1000)
+                    {
+                        eprosima::fastrtps::rtps::IPLocator::setPhysicalPort(server_address, port);
+                        eprosima::fastrtps::rtps::IPLocator::setIPv4(server_address, ip_address);
+                    }
+                    break;
+
+                case optionIndex::UNKNOWN_OPT:
+                    std::cerr << "ERROR: " << opt.name << " is not a valid argument." << std::endl;
+                    option::printUsage(fwrite, stdout, usage, columns);
+                    return 1;
+                    break;
+            }
+        }
+    }
+    else
+    {
+        std::cerr << "ERROR: <publisher|subscriber|server> argument is required." << std::endl;
+        option::printUsage(fwrite, stdout, usage, columns);
+        return 1;
+    }
+
+    // Set default IP address if not specified
+    if (!IsAddressDefined(server_address))
```

Comment:
NIT: Why not setting an int and a string in the arguments, and here set the whole Locator (once the possible errors has been checked)